tests/lib/rules/no-undeclared-export.js

```diff
@@ -21,132 +21,147 @@ var ruleTester = new RuleTester();
+        {
             code: 'export const doc = [window]',
             errors: [
-              {
-                message: 'Export must reference a variable declared in module\'s scope',
-              },
+                {
+                    message: 'Export must reference a variable declared in module\'s scope',
+                },
             ],
-          },
-          {
+        },
+        {
             code: 'export const doc = ["a", [window]]',
             errors: [
-              {
-                message: 'Export must reference a variable declared in module\'s scope',
-              },
+                {
+                    message: 'Export must reference a variable declared in module\'s scope',
+                },
             ],
-          },
-          {
+        },
+        {
             code: 'export const doc = ["a", [{foo: window}]]',
             errors: [
-              {
-                message: 'Export must reference a variable declared in module\'s scope',
-              },
+                {
+                    message: 'Export must reference a variable declared in module\'s scope',
+                },
             ],
-          },
-          {
+        },
+        {
             code: 'const bad = ["a", [{foo: window}]]; export const doc = bad',
             errors: [
-              {
-                message: 'Export must reference a variable declared in module\'s scope',
-              },
+                {
+                    message: 'Export must reference a variable declared in module\'s scope',
+                },
             ],
-          },
-          {
+        },
+        {
             code: `
                 const bad = ["a", [{foo: window}]]; 
                 const fake = bad;
                 export const doc = fake;
             `,
             errors: [
-              {
-                message: 'Export must reference a variable declared in module\'s scope',
-              },
+                {
+                    message: 'Export must reference a variable declared in module\'s scope',
+                },
             ],
-          },
-          {
+        },
+        {
             code: `
                 const a = window;
                 const bad = [a, [{foo: 'foo'}]]; 
                 const fake = bad;
                 export const doc = fake;
             `,
             errors: [
-              {
-                message: 'Export must reference a variable declared in module\'s scope',
-              },
+                {
+                    message: 'Export must reference a variable declared in module\'s scope',
+                },
+            ],
+        },
+        {
+            code: "export const foo = globalThis;",
+            errors: [
+                {
+                    message: 'Export must reference a variable declared in module\'s scope',
+                },
+            ],
+        },
+        {
+            code: "export const foo = () => globalThis;",
```

Comment:
@dejang @erights this is the only one that I could find that is not covered by the rules as today. An array function returning a global binding `globalThis` (which is already implemented in chrome). But it is not about the binding, replacing `globalThis` with `window` is also not covered.